test/k8s-integration/version.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"k8s.io/klog"
+)
+
+var (
+	versionNum           = `(0|[1-9][0-9]*)`
+	internalPatchVersion = `(\-[a-zA-Z0-9_.+-]+)`
+
+	versionRegex         = regexp.MustCompile(`^` + versionNum + `\.` + versionNum + `\.` + versionNum + internalPatchVersion + "?$")
+	gkeExtraVersionRegex = regexp.MustCompile(`^(?:gke)\.(0|[1-9][0-9]*)$`)
+)
+
+type Version struct {
+	version [4]int
+	extra   string
```

Comment:
Does this need to be a struct field?

Reply:
can you elaborate please? array helps in easier range index based comparison.

Reply:
Sorry I meant whether we need `extra` inside the struct rather than a temporary var inside a function